packages/grid/x-data-grid-pro/src/hooks/features/rowPinning/useGridRowPinning.ts

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { useGridApiMethod } from '@mui/x-data-grid';
+import { GridStateInitializer } from '@mui/x-data-grid/internals';
+
+import { GridApiPro } from '../../../models/gridApiPro';
+import { DataGridProProcessedProps } from '../../../models/dataGridProProps';
+import { GridRowPinningApi } from './gridRowPinningInterface';
+
+export const rowPinningStateInitializer: GridStateInitializer<
+  Pick<DataGridProProcessedProps, 'pinnedRows' | 'getRowId'>
+> = (state, props, apiRef) => {
+  apiRef.current.unstable_caches.pinnedRows = {
+    top: [],
+    bottom: [],
+  };
+
+  return {
+    ...state,
+    rows: {
+      ...state.rows,
+      additionalRowGroups: {
+        ...state.rows?.additionalRowGroups,
+        pinnedRows: { top: [], bottom: [] },
+      },
+    },
+  };
+};
+
+export const useGridRowPinning = (
+  apiRef: React.MutableRefObject<GridApiPro>,
+  props: Pick<DataGridProProcessedProps, 'pinnedRows' | 'getRowId' | 'experimentalFeatures'>,
+): void => {
+  const isRowPinned = React.useCallback<GridRowPinningApi['unstable_isRowPinned']>(
+    (rowId) => {
+      return apiRef.current.getRowNode(rowId)?.isPinned || false;
```

Comment:
Not sure this method is worth adding since it can be replaced by a one liner that is not using any internal

Reply:
Agree, and it's only used once 👍